### PR TITLE
Feat/select

### PR DESCRIPTION
**Pull request by brenoliradev**
PR for: https://github.com/posva/pinia-colada/issues/111

### CODE REVIEW

File: src/query-options.ts [hunk 1]
```diff
@@ -155,6 +155,12 @@ export interface UseQueryOptions<TResult = unknown, TError = ErrorDefault> {
    * @default true
    */
   refetchOnReconnect?: MaybeRefOrGetter<_RefetchOnControl>
+
+  /**
+   * Transforms the query data to `TData`
+   * TODO: write better doc
+   */
+  select?: (data: NoInfer<TResult>) => NoInfer<TResult>
```
Comment (brenoliradev) [inline src/query-options.ts:163]: Need help here. Should data be this or we do it as `any` (tanstack query does the latter)? 

Which type would work the best?

File: src/query-store.ts [hunk 2]
```diff
@@ -839,3 +839,23 @@ export function hydrateQueryCache(
 export function serializeQueryCache(queryCache: QueryCache): UseQueryEntryNodeSerialized[] {
   return serializeTreeMap(queryCache.caches)
 }
+
+/**
+ * Initial handler for select
+ */
+const parseSelectedData = <TResult, TError>(select: UseQueryOptions['select'],
```
Comment (brenoliradev) [inline src/query-store.ts:846]: Is it ok to extract and reuse it around? 

It prob need a better jsdoc though

File: src/query-store.ts [hunk 3]
```diff
@@ -839,3 +839,23 @@ export function hydrateQueryCache(
...
+  // NOTE: NoInfer ensures correct inference of TResult and TError
+  state: DataState<NoInfer<TResult>, NoInfer<TError>>): DataState<NoInfer<TResult>, NoInfer<TError>> => {
+    if (typeof select !== 'function' || state.status !== 'success') return state
+
+    try {
+      return { ...state, data: select(state.data) }
+    } catch (e) {
+      // TODO: check how to handle the error state
```
Comment (brenoliradev) [inline src/query-store.ts:854]: Need help here too. Which kinda of error should be written there?

File: src/use-query.spec.ts [hunk 4]
```diff
@@ -1208,4 +1208,60 @@ describe('useQuery', () => {
...
+        throw new Error('Invalid select function')
+      },
+    })
+
+    await flushPromises()
+
+    expect(wrapper.vm.state.error).toBeInstanceOf(Error)
+  })
+
+  // TODO discuss if it should or not
```
Comment (brenoliradev) [inline src/use-query.spec.ts:1255]: Should it updates `initialData`?

**Review by brenoliradev — COMMENTED** [review]
hey @posva when possible give me pointers here! 

best,

**Comment by posva** [comment]
See https://github.com/posva/pinia-colada/discussions/113

**Comment by posva** [comment]
If you want to go forward with this, I think creating plugins for commonly used options would be the way! 👍

**Pull request closed without merging**